File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch06/C/P06-Iter.cpp
// Real root of a real function by the method of successive approximations
#include <stdio.h>
#include <math.h>
#include "roots.h"

double func(double x) { return x - exp(-x); }

int main()
{
   double a, b, x;

   a = -1e10; b = 1e10;                                    // search interval
   x = 0e0;                                          // initial approximation

   if (Iter(func,a,b,x) == 0) {
      printf("x = %8.5f   f(x) = %7.0e\n",x,func(x));
   } else { printf("No solution found !\n"); }
}

File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch06/Python/P06-Iter.py
# Real root of a real function by the method of successive approximations
from math import *
from roots import *

def func(x): return x - exp(-x)

# main

a = -1e10; b = 1e10                                         # search interval
x = 0e0                                               # initial approximation

(x,ierr) = Iter(func,a,b,x)
if (ierr == 0):
   print("x = {0:8.5f}   f(x) = {1:7.0e}".format(x,func(x)))
else: print("No solution found !")
